query.js

```diff
@@ -141,11 +141,13 @@ exports.init = function (sbot, config) {
     hydrateFromMsg(msg, seed) {
       const { feedpurpose, subfeed, nonce } = msg.value.content
       const metadata = self.collectMetadata(msg)
-      const nonceB64 = nonce.toString('base64')
       const keys =
         subfeed === sbot.id
           ? config.keys
-          : sbot.metafeeds.keys.deriveFeedKeyFromSeed(seed, nonceB64)
+          : sbot.metafeeds.keys.deriveFeedKeyFromSeed(
```

Comment:
So this (`hydrateFromMsg()`) will only ever get called with a `msg` that was generated for a new SSB identity? In other words, `nonce` will never be `null` or `undefined` for the L149 call?

Reply:
No, this function will be called for both cases, but in the case of an existing feed, the `nonce` is undefined, so we cannot call `nonce.toString('base64')`

Reply:
That's what I was wondering about. Calling `.toString('base64')` on `null` or `undefined` produces a `TypeError`, right?

Reply:
Oh yeah, it might not be visually obvious, but when `subfeed === sbot.id`, then we assume the case "Existing feed".

Reply:
Ah I understand now. I knew there must be something I was missing. Thanks for explaining.